dali/operators/image/color/color_twist.cc

```diff
@@ -143,15 +152,23 @@ bool ColorTwistCpu::SetupImpl(std::vector<OutputDesc> &output_desc, const HostWo
   const auto &input = ws.template Input<CPUBackend>(0);
   output_desc.resize(1);
   DetermineTransformation(ws);
+  auto sh = input.shape();
+  auto num_dims = sh.sample_dim();
+  auto layout = input.GetLayout();
+  int c_dim = layout.find('C');
+  DALI_ENFORCE(c_dim == num_dims - 1 || layout.empty(), make_string("Only channel last or empty "
+              "layouts are supported, received ", layout, " instead"));
+  DALI_ENFORCE(num_dims >= 3 && num_dims <= 4, make_string("Only 3 and 4 dimensions are "
+              "supported received ", num_dims));
   TYPE_SWITCH(input.type(), type2id, InputType, (uint8_t, int16_t, int32_t, float, float16), (
-      TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float, float16), (
-          {
-              using Kernel = TheKernel<OutputType, InputType>;
-              kernel_manager_.Initialize<Kernel>();
-              auto shapes = CallSetup<Kernel, InputType>(input);
-              output_desc[0] = {shapes, output_type_};
-          }
-      ), DALI_FAIL(make_string("Unsupported output type: ", output_type_)))  // NOLINT
+    TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float, float16), (
+      {
+        using Kernel = TheKernel<OutputType, InputType>;
+        kernel_manager_.Initialize<Kernel>();
+        CallSetup<Kernel, InputType>(input);
```

Comment:
I believe that the point of having that CallKernel specialized for CPU and GPU versions separately (including a basically empty implementation for the CPU) was so that we can have the Setup function shared across the backends. I'd do one of the two:
1. Write a single Setup function
2. Get rid of CallSetup and simplify CPU and GPU implementations accordingly

Reply:
Removed CallSetup. I was not able to unify Setup function keeping TheKernel backend specific.